andela/zinnia-ah-backend
# 165490660 housekeeping - validation and authentication

#### What does this PR do?
Adds validation and authentication to protected routes

#### Description of Task to be completed?
- [x] Validate inputs for request params
- [x] Validate inputs for request body
- [x] Validate inputs for request query

#### How should this be manually tested?
- Clone the repo,
- Run ```npm install``` to install all dependencies,
- Create a .env file using the .env.example as a guide and enter your environment variables
- Ensure you have a PostgreSQL server running on your development environment
- Run `sequelize db:create`  
- Run `sequelize db:migrate` 
- Run `sequelize db:seed:all`   
- Run `npm run start:dev`
- Send a POST to http://localhost:3000/api/v1/articles via your REST API client (postman or insomnia).
- You should see the response below

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?(if applicable)
165490660
<img width="1440" alt="Screenshot 2019-04-24 at 7 08 02 PM" src="https://user-images.githubusercontent.com/12516195/56682943-82777200-66c4-11e9-90c8-2a72ec8ae002.png">
<img width="1440" alt="Screenshot 2019-04-24 at 7 05 01 PM" src="https://user-images.githubusercontent.com/12516195/56682953-86a38f80-66c4-11e9-9213-a7bb4e1204c3.png">


#### Screenshots
